src/samplebuffer.h

```diff
@@ -0,0 +1,73 @@
+#ifndef SAMPLEBUFFER_H
+#define SAMPLEBUFFER_H
+
+#include "util/types.h" // CSAMPLE
+
+#include <algorithm> // std::swap
+
+// A sample buffer with properly aligned memory to enable SSE optimizations.
+// The public interface closely follows that of std::vector.
+//
+// No resize operation is provided intentionally for maximum efficiency!
+// If the size of an existing sample buffer needs to be altered after
+// construction this can simply be achieved by swapping the contents with
+// a temporary sample buffer that has been constructed with the desired
+// size:
+//
+//     SampleBuffer sampleBuffer(oldSize);
+//     ...
+//     SampleBuffer tempBuffer(newSize)
+//     ... copy data from sampleBuffer to tempBuffer...
+//     sampleBuffer.swap(sampleBuffer);
+//
+// The local variable tempBuffer can be omitted if no data needs to be
+// copied from the existing sampleBuffer:
+//
+//     SampleBuffer sampleBuffer(oldSize);
+//     ...
+//     SampleBuffer(newSize).swap(sampleBuffer);
+//
+// After construction the content of the buffer is uninitialized.
+class SampleBuffer {
+    Q_DISABLE_COPY(SampleBuffer);
+
+  public:
+    typedef CSAMPLE value_type;
+    typedef int size_type;
```

Comment:
It will be always a size of a sample buffer or a index of a sample. 
It is useless once we deal with QT containers, so we should probably be more specific. 
CSAMLPE_SIZE -> bad because missleading 
CSAMPLE_INDEX -> quite long and also missleading because it is used for frames as well. 

SINT works for me.